languages/julia/exercises/concept/poetry-club/.meta/design.md

```diff
@@ -0,0 +1,68 @@
+## Learning objectives
+
+- Basic string functions like `titlecase`, `replace` etc.
+- Be aware that unicode strings are the default and that the above mentioned functions will work on non-ascii input.
+- Know that strings can be treated as a collection of characters, e.g. for indexing.
+- How to concatenate strings.
+  - That and why `*` (and not `+`) is used for string concat.
+
+## Out of scope
+
+- Graphemes.
+- Regex.
+- Performance discussions of various ways to construct strings.
+
+## Concepts
+
+The Concepts this exercise unlocks are:
+
+- `strings`:
+
+## Prerequisites
+
+There are no prerequisites.
+
+## Analyzer
+
+- Using `s[1]`/`s[length(s)]` should be discouraged. Suggest `s[begin]`/`s[end]` or `first(s)`/`last(s)` instead.
```

Comment:
`s[length(s)]` can be made to fail, and probably already will fail for some of the inputs below.

`s[lastindex(s)]` would work, but then that's equivalent to `s[end]`.